src/main/java/co/cask/wrangler/internal/TextSpecification.java

```diff
@@ -81,83 +86,158 @@ public TextSpecification(String dsl) {
     for (String line : lines) {
-      line = line.trim().replaceAll(" +"," ");
-      String[] options = line.split(" ",0);
-      String command = options[0];
-      String qualifier = options[1];
+      StringTokenizer tokenizer = new StringTokenizer(line, " ");
+      String command = tokenizer.nextToken();
 
       switch (command) {
-        case "set":
-          switch (qualifier) {
+        case "set": {
+          switch (tokenizer.nextToken()) {
             // set format [csv|json] <delimiter> <skip empty lines>
-            case "format":
-              if (options[2].equalsIgnoreCase("csv")) {
-                boolean ignoreEmptyLines = false;
-                if (options[3].equalsIgnoreCase("true")) {
-                  ignoreEmptyLines = true;
-                }
-                char delimiter = options[3].charAt(0);
-                if (options[3].startsWith("\\")) {
-                  String unescapedStr = StringEscapeUtils.unescapeJava(options[3]);
+            case "format": {
+              if (tokenizer.nextToken().equalsIgnoreCase("csv")) {
+                String delimStr = tokenizer.nextToken();
+                char delimiter = delimStr.charAt(0);
+                if (delimStr.startsWith("\\")) {
+                  String unescapedStr = StringEscapeUtils.unescapeJava(delimStr);
                   if (unescapedStr == null) {
-                    throw new IllegalArgumentException("Invalid delimiter for CSV Parser: " + options[3]);
+                    throw new IllegalArgumentException("Invalid delimiter for CSV Parser: " + delimStr);
                   }
                   delimiter = unescapedStr.charAt(0);
                 }
+                boolean ignoreEmptyLines = false;
+                if (tokenizer.nextToken().equalsIgnoreCase("true")) {
+                  ignoreEmptyLines = true;
+                }
                 CsvParser.Options opt = new CsvParser.Options(delimiter, ignoreEmptyLines);
                 steps.add(new CsvParser(lineno, line, opt, STARTING_COLUMN, false));
                 steps.add(new Drop(lineno, line, STARTING_COLUMN));
               } else {
-                throw new ParseException("Unknown format " + options[3], lineno);
+                throw new ParseException("Unknown format ", lineno);
               }
-              break;
+            }
+            break;
+
+            // set column <column-name> <jexl-expression>
+            case "column": {
+              String column = tokenizer.nextToken();
+              String expr = tokenizer.nextToken("\n");
+              steps.add(new Expression(lineno, line, column, expr));
+            }
+            break;
 
             // set columns <name1, name2, ...>
-            case "columns":
-              String cols[] = options[2].split(",");
+            case "columns": {
+              String cols[] = tokenizer.nextToken().split(",");
               steps.add(new Columns(lineno, line, Arrays.asList(cols)));
```

Comment:
```replaceColumnNames``` is true, so it will replace it. That's only functionality that is needed. What's the use case you are thinking that would require merging ? @abhinavbansal246